src/App.tsx

```diff
@@ -1,61 +1,115 @@
 import './App.scss';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { getTodoId, getUserById, handleChanges } from './helper';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [currentTodos, setCurrentTodos] = useState(todos);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [newTodoUserId, setNewTodoUserId] = useState('');
+  const newTodoUser = usersFromServer
+    .find(user => user.id === +newTodoUserId);
```

Comment:
Why are you looking for a user on every render?

Move it to moment when you create new todo
And you have a handler `getUserById` which can help you